### ISSUE TITLE
iconOffset is ignored with Expression.step and zoom()

### ISSUE THREAD

**Issue by sigis151** [mapbox/mapbox-gl-native#11498]
**Platform:** Android 
**Mapbox SDK version:** 6.0.0-beta.3, 6.0.0-beta.4

### Steps to trigger behavior
Add iconOffset property to `SymbolLayer`:
```kotlin
iconOffset(Expression.step(zoom(), literal(arrayOf(0f, 0f)),
  stop(1, arrayOf(0f, 0f)),
  stop(10, arrayOf(0f, -35f))
))
```
### Expected behavior
Functionality should be the same as with functions. The icon should be offset when zoom level is >=10 :
```kotlin
iconOffset(Function.zoom(IntervalStops.interval(
  Stop.stop(1, iconOffset(arrayOf(0f, 0f))),
  Stop.stop(10, iconOffset(arrayOf(0f, -35f)))
)))
```
### Actual behavior
offset is ignored using Expressions API.

**Additional notes**:
iconImage property works fine:
```kotlin
iconImage(step(zoom(), literal("marker_small"),
  stop(1, "marker_small"),
  stop(10, "marker_big")
))
```

 `Expression.step` requires three parameters(`input` `expression` `stops`) but documentation doesn't provide description for second parameter

**Comment by tobrun** [comment]
@sigis151 thank you for reaching out and trying out expressions! Would you able to share the signature of the `arrayOf` method?  I was able to get it working with the following:

```java
iconOffset(step(zoom(), literal(new Float[] {0f, 0f}),
  literal(1), literal(new Float[] {0f, 0f}),
  literal(10), literal(new Float[] {0f, -35f})
)),
```

![ezgif com-video-to-gif 55](https://user-images.githubusercontent.com/2151639/37771662-a7672bf0-2dae-11e8-9554-f8b017580ff7.gif)

While reproducing I hit a couple of issues myself which I will address in a PR.
Thanks again for trying out expressions!

**Comment by tobrun** [comment]
Currently working on correctly integrating `float[]`, it's currently being parsed as an object instead of an Array. Workaround for now is to use `Float[]` instead.

**Comment by sigis151** [comment]
For me it would be more intuitive to use `PointF` or any alternative to define offset position rather than float array

**Comment by tobrun** [comment]
landed with #11500
